p8e-api/src/main/kotlin/io/provenance/engine/service/ChaincodeInvokeService.kt

```diff
@@ -11,21 +14,33 @@ import io.p8e.util.toUuidProv
 import io.p8e.util.configureProvenance
 import io.provenance.engine.config.ChaincodeProperties
 import io.provenance.engine.crypto.Account
+import io.provenance.engine.crypto.Bech32
 import io.provenance.engine.domain.TransactionStatusRecord
+import io.provenance.engine.util.toProv
 import io.provenance.metadata.v1.MsgAddP8eContractSpecRequest
+import io.provenance.metadata.v1.MsgP8eMemorializeContractRequest
 import io.provenance.p8e.shared.domain.ContractTxResult
 import io.provenance.pbc.clients.*
-import io.provenance.pbc.clients.p8e.changeScopeOwnership
-import io.provenance.pbc.clients.p8e.memorializeP8EContract
 import io.provenance.pbc.clients.tx.BatchTx
-import io.provenance.pbc.clients.tx.TxPreparer
-import io.provenance.pbc.clients.tx.batch
+import org.bouncycastle.crypto.digests.RIPEMD160Digest
 import org.jetbrains.exposed.sql.transactions.transaction
 import org.springframework.stereotype.Component
 import java.util.*
 import java.util.concurrent.CompletableFuture
-import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.Future
+import java.util.concurrent.LinkedBlockingQueue
+import java.util.concurrent.TimeUnit
 import kotlin.concurrent.thread
+import kotlin.random.Random
+
+const val SIGNATURE_VERIFICATION_FAILED = 4
+
+data class ContractRequestWrapper(
+    val uuid: UUID,
```

Comment:
maybe should rename this executionUuid, since that appears to be what we are putting here... seems like we have too many uuids to not be specific =)